src/main/java/HttpServer/MockTraffic.java

```diff
@@ -32,4 +32,9 @@ public String readLine() {
         return request.remove(0);
     }
 
+    public void emulateSimplePut() {
+        request = new ArrayList<>(1);
+        request.add("PUT /form HTTP/1.1");
```

Comment:
Most of your tests at the unit level should be creating request/response domain objects directly, and not parsing them from strings. (Obviously, not the ones that are testing the parsing logic).

We can chat about this in person.